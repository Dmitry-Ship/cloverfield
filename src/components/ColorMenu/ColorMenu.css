

.colorMenu {
	position: relative;
}

.colorMenu:hover .popupMenu {
	display: block;
}

.palette {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-around;
  width: 150px;
  height: auto;
}

.circle {
	margin: 4px 4px;
  width: 28px;
  height: 28px;
  border-radius: 50%;
	cursor: pointer;
	transition: all 300ms ease-in-out;
}

.circle:first-child {
	border: #808080;
}

.circle:hover {
	border: 2px solid var(--color__flat__wet_asphalt);
}

.chosen {
	border: 2px solid #000;
}

.popupMenu {
  position: absolute;
	bottom: 100%;
	left: 50%;
	display: none;
  background: white;
	transform: translateX(-50%);
}

.red {
	background: var(--color__red);
	composes: circle;
}

.orange {
	background: var(--color__orange);
	composes: circle;
}

.yellow {
	background: var(--color__yellow);
	composes: circle;
}

.grey {
	background: var(--color__grey);
	composes: circle;
}

.blue {
	background: var(--color__blue);
	composes: circle;
}

.teal {
	background: var(--color__teal);
	composes: circle;
}

.green {
	background: var(--color__green);
	composes: circle;
}

.white {
	background: #fff;
	composes: circle;
}
