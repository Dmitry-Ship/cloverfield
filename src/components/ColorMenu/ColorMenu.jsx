import React, { PropTypes } from 'react';

import styles from './ColorMenu.scss';

import PopUpMenu from '../basic/PopUpMenu';
import Icon from '../basic/Icon';

const ColorMenu = ({ color, onSetColor, className, position }) => {
  const colors = ['white', 'red', 'orange', 'yellow', 'grey', 'blue', 'teal', 'green'];

  const circles = colors.map((item, i) => {
    const chosen = item === color ? styles.chosen : '';
    return (
      <span
        onClick={() => onSetColor(item)}
        key={i}
        className={`${styles[item]} ${chosen}`}
      />
    );
  });

  return (
    <Icon className={`${styles.colorMenu} ${className}`} name="palette" >
      <PopUpMenu position={position} className={styles.popupMenu} >
        <div className={styles.palette}>
          {circles}
        </div>
      </PopUpMenu>
    </Icon>
  );
};

export default ColorMenu;

ColorMenu.defaultProps = {
  color: 'white',
  className: '',
  position: 'top',
};

ColorMenu.propTypes = {
  color: PropTypes.string.isRequired,
  onSetColor: PropTypes.func.isRequired,
  className: PropTypes.string,
  position: PropTypes.oneOf(['top', 'buttom', 'left', 'right']),
};
