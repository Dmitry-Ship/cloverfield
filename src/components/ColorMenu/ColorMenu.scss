@value colors: "../../styles/colors.scss";
@value flat_wet_asphalt, note_green, note_red, note_yellow, note_blue, note_grey, note_orange, note_teal from colors;

.colorMenu {
  position: relative;
  transform: translateX(25%);
  &:hover .popupMenu {
    display: block;
  }
}

.palette {
  width: 150px;
  height: auto;
  flex-wrap: wrap;
  justify-content: space-around;
  display: flex;
}

.circle {
  border-radius: 50%;
  width: 28px;
  height: 28px;
  margin: 4px 4px;
  cursor: pointer;
  &:first-child {
    border-color: grey;
  }
  &:hover {
    border: 2px solid flat_wet_asphalt;
  }
}

.chosen {
  composes: circle;
  border: 2px solid black;
}

.red {
  composes: circle;
  background: note_red;
}

.orange {
  composes: circle;
  background: note_orange;
}

.yellow {
  composes: circle;
  background: note_yellow;
}

.grey {
  composes: circle;
  background: note_grey;
}

.blue {
  composes: circle;
  background: note_blue;
}

.teal {
  composes: circle;
  background: note_teal;
}

.green {
  composes: circle;
  background: note_green;
}

.white {
  composes: circle;
  border: 2px solid grey;
  background: white;
}