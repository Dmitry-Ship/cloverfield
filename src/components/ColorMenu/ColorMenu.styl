
@import "../../styles/mixins.styl"
@value colors "../../styles/colors.styl";
@value flat_wet_asphalt from colors;
@value note_green from colors;
@value note_red from colors;
@value note_yellow from colors;
@value note_blue from colors;
@value note_grey from colors;
@value note_orange from colors;
@value note_teal from colors;

.colorMenu
  position relative
  &:hover
    .popupMenu
      display block

.palette
  size 150px auto
  flex-wrap wrap
  justify-content space-around
  display flex

.circle
  circle(28px)
  margin 4px 4px
  transition all 300ms ease-in-out
  cursor pointer
  &:first-child
    border grey
  &:hover
    border 2px solid flat_wet_asphalt

.chosen
  composes: circle
  border 2px solid black

.popupMenu
  display none
  tooltip(white, 10px, bottom)
  transform translateX(-50%)
  bottom calc(100% + 10px)
  left 50%

.red
  composes circle
  background note_red
.orange
  composes circle
  background note_orange
.yellow
  composes circle
  background note_yellow
.grey
  composes circle
  background note_grey
.blue
  composes circle
  background note_blue
.teal
  composes circle
  background note_teal
.green
  composes circle
  background note_green
.white
  composes circle
  background white
