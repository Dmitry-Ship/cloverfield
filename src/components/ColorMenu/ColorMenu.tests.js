import React              from 'react';
import { Random }         from 'meteor/random';
import { shallow, mount } from 'enzyme';
import StubCollections    from 'meteor/hwillson:stub-collections';
import { sinon }          from 'meteor/practicalmeteor:sinon';
import { expect }         from 'chai';
import { Notes }          from '../../../api/notes/notes.js';

import SetColorMenu       from './SetColorMenu.jsx';

if (Meteor.isClient) {
  describe('SetColorMenu', () => {
    let note;
    beforeEach(function() {
      StubCollections.stub(Notes);

      Factory.define('note', Notes);

      note = Factory.build('note', { color: 'white'});
    })

    afterEach(function() {
      StubCollections.restore();
    })


    it('renders correctly', () => {
      const wrapper = mount(<SetColorMenu color={note.color} />);
      const circles = wrapper.find('span');

      expect(circles).to.have.length(8);
      expect(circles.at(0).hasClass('chosen')).to.be.true;
    });

    it('changes color', () => {
      //sinon.stub(setColor, 'call');

      const spy = sinon.spy();
      const wrapper = mount(<SetColorMenu color={note.color} onSetColor={spy}/>);
      const circles = wrapper.find('span');

      for (let i = 0; i < circles.length; i++) {
        const currentCircle = circles.at(i), // at returns node wrapped in enzyme stuff
              currentColor = circles.get(i).className.split(' ')[0]; // get returns just a node
        currentCircle.simulate('click');
        sinon.assert.calledWith(spy, currentColor);

        //expect(currentCircle.hasClass('chosen')).to.be.true;

        // sinon.assert.calledWith(setColor.call, {
        //   noteId: note._id,
        //   color: currentColor,
        // });
      }

      //setColor.call.restore();
    });
  })
}
