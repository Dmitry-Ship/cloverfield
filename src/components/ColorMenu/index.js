export { default } from './ColorMenu.jsx';
