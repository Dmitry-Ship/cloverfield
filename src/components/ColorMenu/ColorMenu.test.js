import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ColorMenu from './ColorMenu';

describe('<ColorMenu />', () => {
  const onSetColor = jest.fn();

  it('should render correctly', () => {
    const tree = renderer.create(
      <ColorMenu onSetColor={onSetColor} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when position passed in', () => {
    const tree = renderer.create(
      <ColorMenu position="top" onSetColor={onSetColor} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when color passed in', () => {
    const tree = renderer.create(
      <ColorMenu color="red" onSetColor={onSetColor} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onSetColor with right argument', () => {
    const wrapper = shallow(
      <ColorMenu onSetColor={onSetColor} />
    );

    wrapper.find('span').first().simulate('click');

    expect(onSetColor.mock.calls[0][0]).toBe('white');
  });
});
