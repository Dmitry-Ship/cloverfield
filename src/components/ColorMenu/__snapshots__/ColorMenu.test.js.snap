// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorMenu /> should render correctly 1`] = `
<i
  className="colorMenu  material-icons"
>
  palette
  <div
    className="top popupMenu"
  >
    <div
      className="palette"
    >
      <span
        className="white chosen"
        onClick={[Function]}
      />
      <span
        className="red "
        onClick={[Function]}
      />
      <span
        className="orange "
        onClick={[Function]}
      />
      <span
        className="yellow "
        onClick={[Function]}
      />
      <span
        className="grey "
        onClick={[Function]}
      />
      <span
        className="blue "
        onClick={[Function]}
      />
      <span
        className="teal "
        onClick={[Function]}
      />
      <span
        className="green "
        onClick={[Function]}
      />
    </div>
  </div>
</i>
`;

exports[`<ColorMenu /> should render correctly when color passed in 1`] = `
<i
  className="colorMenu  material-icons"
>
  palette
  <div
    className="top popupMenu"
  >
    <div
      className="palette"
    >
      <span
        className="white "
        onClick={[Function]}
      />
      <span
        className="red chosen"
        onClick={[Function]}
      />
      <span
        className="orange "
        onClick={[Function]}
      />
      <span
        className="yellow "
        onClick={[Function]}
      />
      <span
        className="grey "
        onClick={[Function]}
      />
      <span
        className="blue "
        onClick={[Function]}
      />
      <span
        className="teal "
        onClick={[Function]}
      />
      <span
        className="green "
        onClick={[Function]}
      />
    </div>
  </div>
</i>
`;

exports[`<ColorMenu /> should render correctly when position passed in 1`] = `
<i
  className="colorMenu  material-icons"
>
  palette
  <div
    className="top popupMenu"
  >
    <div
      className="palette"
    >
      <span
        className="white chosen"
        onClick={[Function]}
      />
      <span
        className="red "
        onClick={[Function]}
      />
      <span
        className="orange "
        onClick={[Function]}
      />
      <span
        className="yellow "
        onClick={[Function]}
      />
      <span
        className="grey "
        onClick={[Function]}
      />
      <span
        className="blue "
        onClick={[Function]}
      />
      <span
        className="teal "
        onClick={[Function]}
      />
      <span
        className="green "
        onClick={[Function]}
      />
    </div>
  </div>
</i>
`;
